logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/sessionreset/ResetSessionUseCaseTest.kt

```diff
@@ -0,0 +1,208 @@
+
+        given(sessionResetSender)
+            .suspendFunction(sessionResetSender::invoke)
+            .whenInvokedWith(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .thenReturn(Either.Right(Unit))
+
+        given(messageRepository)
+            .suspendFunction(messageRepository::markMessagesAsDecryptionResolved)
+            .whenInvokedWith(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .thenReturn(Either.Left(failure))
+
+        val result = resetSessionUseCase(CONVERSATION_ID, USER_ID, CLIENT_ID)
+
+        verify(idMapper)
+            .function(idMapper::toCryptoQualifiedIDId)
+            .with(anything())
+            .wasInvoked(exactly = once)
+
+        verify(proteusClient)
+            .function(proteusClient::deleteSession)
+            .with(anything())
+            .wasInvoked(exactly = once)
+
+        verify(messageRepository)
+            .function(messageRepository::markMessagesAsDecryptionResolved)
+            .with(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .wasInvoked(exactly = once)
+
+        assertEquals(ResetSessionResult.Failure(failure), result)
+    }
+
+    @Test
+    fun givenResetSessionCalled_whenRunningSuccessfully_thenReturnSuccessResult() = runTest(testDispatchers.io) {
+        given(proteusClientProvider)
+            .suspendFunction(proteusClientProvider::getOrError)
+            .whenInvoked()
+            .thenReturn(Either.Right(proteusClient))
+
+        given(idMapper)
+            .function(idMapper::toCryptoQualifiedIDId)
+            .whenInvokedWith(eq(USER_ID))
+            .thenReturn(CRYPTO_USER_ID)
+
+        given(sessionResetSender)
+            .suspendFunction(sessionResetSender::invoke)
+            .whenInvokedWith(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .thenReturn(Either.Right(Unit))
+
+        given(messageRepository)
+            .suspendFunction(messageRepository::markMessagesAsDecryptionResolved)
+            .whenInvokedWith(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .thenReturn(Either.Right(Unit))
+
+        val result = resetSessionUseCase(CONVERSATION_ID, USER_ID, CLIENT_ID)
+
+        verify(idMapper)
+            .function(idMapper::toCryptoQualifiedIDId)
+            .with(anything())
+            .wasInvoked(exactly = once)
+
+        verify(proteusClient)
+            .function(proteusClient::deleteSession)
+            .with(anything())
+            .wasInvoked(exactly = once)
+
+        verify(messageRepository)
+            .function(messageRepository::markMessagesAsDecryptionResolved)
+            .with(eq(CONVERSATION_ID), eq(USER_ID), eq(CLIENT_ID))
+            .wasInvoked(exactly = once)
+
+        assertEquals(ResetSessionResult.Success, result)
+
+    }
+
+    companion object {
+        val CONVERSATION_ID = ConversationId("conversation-id", "domain")
+        val CLIENT_ID = ClientId("client-id")
+        val USER_ID = UserId("client-id", "domain")
```

Comment:
I think we can reuse the nice stubs we have on `com.wire.kalium.logic.framework`  ie: `TestClient`